parachains/pallets/bridge-transfer/src/lib.rs

```diff
@@ -0,0 +1,1596 @@
+			let origin_location = T::TransferAssetOrigin::ensure_origin(origin)?;
+
+			// Check remote destination + bridge_config
+			let (_, bridge_config, remote_destination) =
+				Self::ensure_remote_destination(*destination)?;
+
+			// Check assets (lets leave others checks on `AssetTransactor`)
+			let assets: MultiAssets =
+				(*assets).try_into().map_err(|()| Error::<T>::InvalidAssets)?;
+			ensure!(
+				assets.len() <= T::MaxAssetsLimit::get() as usize,
+				Error::<T>::MaxAssetsLimitReached
+			);
+
+			// Do this in transaction (explicitly), the rollback should occur in case of any error and no assets will be trapped or lost
+			Self::do_reserve_and_send_in_transaction(
+				origin_location,
+				remote_destination,
+				assets,
+				bridge_config,
+			)
+		}
+
+		/// Transfer `ping` via bridge to different global consensus.
+		///
+		/// - can be used for testing purposes that bridge transfer is working and configured for `destination`
+		///
+		/// Parameters:
+		///
+		/// * `destination`: Different consensus location, e.g. Polkadot's Statemint: `2, X2(GlobalConsensus(NetworkId::Polkadot), Parachain(1000))`
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::ping_via_bridge())]
+		pub fn ping_via_bridge(
+			origin: OriginFor<T>,
+			destination: Box<VersionedMultiLocation>,
+		) -> DispatchResult {
+			let origin_location = T::TransferPingOrigin::ensure_origin(origin)?;
+
+			// Check remote destination + bridge_config
+			let (network, bridge_config, remote_destination) =
+				Self::ensure_remote_destination(*destination)?;
+
+			// Check reserve account - sovereign account of bridge
+			let allowed_target_location = bridge_config.allowed_target_location;
+
+			// Prepare `ping` message
+			let xcm: Xcm<()> =
+				T::PingMessageBuilder::try_build(&origin_location, &network, &remote_destination)
+					.ok_or(Error::<T>::UnsupportedPing)?;
+
+			// Initiate bridge transfer
+			Self::initiate_bridge_transfer(allowed_target_location, xcm).map_err(Into::into)
+		}
+
+		/// Adds new bridge configuration, which allows transfer to this `bridged_network`.
+		///
+		/// Parameters:
+		///
+		/// * `bridged_network`: Network where we want to allow transfer funds
+		/// * `bridge_config`: contains location for BridgeHub in our network + fee
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::add_exporter_config())]
+		pub fn add_exporter_config(
+			origin: OriginFor<T>,
+			bridged_network: NetworkId,
+			bridge_config: Box<BridgeConfig>,
+		) -> DispatchResult {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
+			ensure!(
+				!AllowedExporters::<T>::contains_key(bridged_network),
+				Error::<T>::InvalidConfiguration
+			);
+			let allowed_target_location_network = bridge_config
+				.allowed_target_location
+				.interior()
+				.global_consensus()
+				.map_err(|_| Error::<T>::InvalidConfiguration)?;
+			ensure!(
+				bridged_network == allowed_target_location_network,
+				Error::<T>::InvalidConfiguration
+			);
```

Comment:
Nit: This seems unneeded. We could just extract the bridged network from the `bridge_config.allowed_target_location` and remove the `bridged_network` parameter.